File: 232. Implement Queue using Stacks.cpp
#include <stack>
class Queue {
private:
    stack <int> s1;
    stack <int> s2;
public:
    // Push element x to the back of queue.
    void push(int x) {
        s1.push(x);
    }

    // Removes the element from in front of queue.
    void pop(void) {
        if (s2.size() == 0)
        {
            while (s1.size() > 0)
            {
                s2.push(s1.top());
                s1.pop();
            }
        }
        s2.pop();
    }

    // Get the front element.
    int peek(void) {
        if (s2.size() == 0)
        {
            while (s1.size() > 0)
            {
                s2.push(s1.top());
                s1.pop();
            }
        }
        return s2.top();
    }

    // Return whether the queue is empty.
    bool empty(void) {
        return s1.empty() && s2.empty() ? true : false;
    }
};

File: 232. Implement Queue using Stacks.py
class Queue(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self.stack1 = []
        self.stack2 = []
        

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        self.stack1 += [x]
        

    def pop(self):
        """
        :rtype: nothing
        """
        if len(self.stack2) == 0:
            while len(self.stack1) > 0:
                self.stack2 += [self.stack1.pop()]
        self.stack2.pop()

    def peek(self):
        """
        :rtype: int
        """
        if len(self.stack2) == 0:
            while len(self.stack1) > 0:
                self.stack2 += [self.stack1.pop()]
        return self.stack2[-1]

    def empty(self):
        """
        :rtype: bool
        """
        return True if len(self.stack1) == 0 and len(self.stack2) == 0 else False
